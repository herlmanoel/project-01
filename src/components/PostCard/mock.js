/*
    Objeto falso para testar o componente
*/
const postCardPropsMock =  {
    title: 'title 1',
    cover: 'img/img.png',
    id: 1,
    body: 'body 1',
}

export { postCardPropsMock };