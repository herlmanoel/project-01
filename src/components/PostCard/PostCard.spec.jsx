import { render, screen } from "@testing-library/react";
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

/*
    Dando um console.log para ver o que está acontecendo
        const { debug } = render(<PostCard post={mock} />);
        debug();
    A imagem não precisa existir

    Estes testes são chamados de snapshot

    Não temos getByRole para parágrafos

    Quando 'termino' um componente e
    se eu mudar alguma coisa no componente
    como eu testo?
    Quando terminamos, posso fazer um teste de snapshot
    tira uma foto do componente do jeito que ele está.

    snapshot: pega o componente, renderiza e salva.
    PostCard.spec.jsx.snap

    Com isso, se eu alterar o componente ele sempre verificará se
    bate com o snapshot, se não bater o teste irá falhar e,
    se for intencional a mudança, posso clicar "u" para ele atualizar
    o snapshot.
*/

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {

        render(<PostCard post={postCardPropsMock} />);

        const img = screen.getByRole('img');
        const heading = screen.getByRole('heading', { name: 'title 1'});
        const paragraph = screen.getByText('body 1');

        expect(img).toBeInTheDocument();
        expect(img).toHaveAttribute('src', postCardPropsMock.cover);
        
        expect(heading).toBeInTheDocument();
        expect(paragraph).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard post={postCardPropsMock} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});