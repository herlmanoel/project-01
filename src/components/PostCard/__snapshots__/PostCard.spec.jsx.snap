// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    class="post__img"
    src="img/img.png"
  />
  <div
    class="post__content"
  >
    <h1
      name="title 1"
    >
       
      title 1
       
    </h1>
    <p>
       
      body 1
       
    </p>
  </div>
</div>
`;
